strawberry/union.py

```diff
@@ -53,11 +56,16 @@ def __init__(
         type_annotations: Tuple[StrawberryAnnotation, ...] = tuple(),
         description: Optional[str] = None,
         directives: Iterable[object] = (),
+        # used for better error messages
+        _source_file: Optional[str] = None,
+        _source_line: Optional[int] = None,
```

Comment:
I personally would prefer to not have those in the `__init__`, since it is mostly for internal usage and it will pollute code completions for people not working in the internals.

If we must have this it would at least be nice to add those after a `*` to make sure they are forced to be passed as keywords, otherwise adding an extra arg in here would either break some APIs or we would be forced to add those after `_source_file`/`_source_line` and make it mandatory to pass those.

But maybe we could just set those in the __init__ like:

```python
self.source_file = Optional[str] = None
self.source_line = Optional[int] = None
```

And set them by hand after instantializing the object.

Also, don't know if this makes sense, but since this type inherits from `StrawberryType`, and a lot of other subclasses might want to use something similar (e.g. `StrawberryList`, `StrawberryOptional`, etc), we could add those lines there so that all `StrawberryType` subclasses can have a way to provide info for rich exceptions?

cc @patrick91

Reply:
the `_source_file` is only for things that don't have an origin so not sure if it makes sense to add it everywhere :)

I like your suggestion, I'll try to implement it today